src/Framework/EngineServices.cs

```diff
@@ -45,5 +53,11 @@ public abstract class EngineServices
         /// This is a performance optimization allowing tasks to skip expensive double-logging.
         /// </remarks>
         public virtual bool IsTaskInputLoggingEnabled => throw new NotImplementedException();
+
+        /// <summary>
+        /// Reports a file access from a task.
+        /// </summary>
+        /// <param name="fileAccessData">The file access to report.</param>
+        public virtual void ReportFileAccess(FileAccessData fileAccessData) => throw new NotImplementedException();
```

Comment:
I don't believe this is correct as this would be a breaking change. I believe you have to add another `IBuildEngineX` or whatever

Reply:
Oh, no maybe I'm wrong. I don't think I understand this then tbh.